homeassistant/components/goodwe/button.py

```diff
@@ -0,0 +1,84 @@
+"""GoodWe PV inverter selection settings entities."""
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class GoodweButtonEntityDescriptionRequired:
+    """Required attributes of GoodweButtonEntityDescription."""
+
+    action: Callable[[Inverter], Awaitable[None]]
+
+
+@dataclass
+class GoodweButtonEntityDescription(
+    ButtonEntityDescription, GoodweButtonEntityDescriptionRequired
+):
+    """Class describing Goodwe button entities."""
+
+
+SYNCHRONIZE_CLOCK = GoodweButtonEntityDescription(
+    key="synchronize_clock",
+    name="Synchronize inverter clock",
+    icon="mdi:clock-check-outline",
+    entity_category=EntityCategory.CONFIG,
+    action=lambda inv: inv.write_setting("time", datetime.now()),
```

Comment:
Is this call really non-blocking? if write_setting perform any io, it should be executed in an exector.

Reply:
It invokes asyncio method without return value

Reply:
I am not sure i like returning the awaitable that way. But i suppose it might work. I would have just used a standard async function instead of a lambda. Ok.